zerver/tests/test_subs.py

```diff
@@ -416,6 +416,22 @@ def test_admin_remove_others_from_unsubbed_private_stream(self):
         self.assert_json_error(
             result, "Cannot administer invite-only streams this way")
 
+    def test_permission_of_stream_creator(self):
+        # type: () -> None
+        """
+        A user who creates a stream should have
+        all permissions in that stream.
+        """
+        email = 'hamlet@zulip.com'
+        self.login(email)
+        user_profile = get_user_profile_by_email(email)
+
+        stream_name = u"new_stream"
+        stream, _ = create_stream_if_needed(user_profile.realm, stream_name)
+        bulk_add_subscriptions([stream], [user_profile], current_user=user_profile)
+        sub = get_subscription(stream_name, user_profile)
+        self.assertEqual(len(sub.permissions_list()), 4)
```

Comment:
It's probably better to explicitly check that each permission in `Stream.PERMISSION_FLAGS` is satisfied here. I can imagine a situation where new permissions are added in the future (and so 4 needs to be incremented), but a bug exists that causes this test to pass anyway.